interfaces/builtin/serial_port.go

```diff
@@ -206,9 +209,69 @@ func (iface *serialPortInterface) AutoConnect(*snap.PlugInfo, *snap.SlotInfo) bo
 	return true
 }
 
+func writeToKey(key hash.Hash, name, val string) {
+	key.Write([]byte(name))
+	key.Write([]byte{0})
+	key.Write([]byte(val))
+	key.Write([]byte{0})
+}
+
+func (iface *serialPortInterface) HotplugKey(di *hotplug.HotplugDeviceInfo) (snap.HotplugKey, error) {
```

Comment:
IMO we need to think of extracting some of this implementation into helpers as much of this is duplicates the logic from hotplug.go.